src/main/java/teammates/sqllogic/core/FeedbackResponsesLogic.java

```diff
@@ -183,5 +188,138 @@ public boolean areThereResponsesForQuestion(UUID questionId) {
+     */
+    public void updateFeedbackResponsesForDeletingStudent(String courseId) {
+        updateRankRecipientQuestionResponsesAfterDeletingStudent(courseId);
+    }
+
+    private void updateRankRecipientQuestionResponsesAfterDeletingStudent(String courseId) {
+        List<FeedbackQuestion> filteredQuestions =
+                fqLogic.getFeedbackQuestionForCourseWithType(courseId, FeedbackQuestionType.RANK_RECIPIENTS);
+        SqlCourseRoster roster = new SqlCourseRoster(
+                usersLogic.getStudentsForCourse(courseId),
+                usersLogic.getInstructorsForCourse(courseId));
+
+        for (FeedbackQuestion question : filteredQuestions) {
+            makeRankRecipientQuestionResponsesConsistent(question, roster);
+        }
     }
+
+    /**
+     * Makes the rankings by one giver in the response to a 'rank recipient question' consistent, after deleting a
+     * student.
+     * <p>
+     *     Fails silently if the question type is not 'rank recipient question'.
+     * </p>
+     */
+    private void makeRankRecipientQuestionResponsesConsistent(
+            FeedbackQuestion question, SqlCourseRoster roster) {
+        if (!question.getQuestionDetailsCopy().getQuestionType().equals(FeedbackQuestionType.RANK_RECIPIENTS)) {
+            return;
+        }
+
+        FeedbackParticipantType giverType = question.getGiverType();
+        List<FeedbackResponse> responses;
+
+        int numberOfRecipients;
+
+        switch (giverType) {
+        case INSTRUCTORS:
+        case SELF:
+            for (Instructor instructor : roster.getInstructors()) {
+                numberOfRecipients =
+                        fqLogic.getRecipientsOfQuestion(question, instructor, null, roster).size();
+                responses = getFeedbackResponsesFromGiverForQuestion(question.getId(), instructor.getEmail());
+
+                FeedbackRankRecipientsResponseDetails
+                        .getUpdatedResponsesForRankRecipientQuestionsMigrated(responses, numberOfRecipients);
+            }
+            break;
+        case TEAMS:
+        case TEAMS_IN_SAME_SECTION:
+            Student firstMemberOfTeam;
+            String team;
+            Map<String, List<Student>> teams = roster.getTeamToMembersTable();
+            for (Map.Entry<String, List<Student>> entry : teams.entrySet()) {
+                team = entry.getKey();
+                firstMemberOfTeam = entry.getValue().get(0);
+                numberOfRecipients =
+                        fqLogic.getRecipientsOfQuestion(question, null, firstMemberOfTeam, roster).size();
+                responses =
+                        getFeedbackResponsesFromTeamForQuestion(
+                                question.getId(), question.getCourseId(), team, roster);
+
+                FeedbackRankRecipientsResponseDetails
+                        .getUpdatedResponsesForRankRecipientQuestionsMigrated(responses, numberOfRecipients);
```

Comment:
1. This should not be static method of response details class, this should be a private method in the same class.
2. Name should not start with get, this is not a getter method. This method should be updating the relevant responses instead.